### PR TITLE
esp32s2: esp_error: Consistent error checking of esp-idf calls

### PR DESCRIPTION
**Pull request by jepler**
I wrote this code for audioio but that's not ready for inclusion, and @microdev1 says it would be useful in their work.

### CODE REVIEW

**Comment by microDev1** [comment]
Thanks! @jepler. One question, what parameters are you using to categorize an error of type `ESP_ERR_` to be eligible for inclusion.

**Comment by jepler** [comment]
@microDev1 I think that any error that would plausibly arise from user code we would want to include.  I just did the ones in the "main" error header, **esp_err.h**.

File: ports/esp32s2/esp_error.c [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+#include "py/runtime.h"
+
+void raise_esp_error(esp_err_t err) {
+    const compressed_string_t *msg = NULL;
+    switch(err) {
+        case ESP_FAIL:
+            msg = translate("Generic Failure");
+            break;
+        case ESP_ERR_NO_MEM:
+            msg = translate("Out of memory");
```
Comment (tannewt) [inline ports/esp32s2/esp_error.c:37]: We have a special subclass of MemoryError for this already: https://github.com/adafruit/circuitpython/blob/main/ports/esp32s2/bindings/espidf/__init__.c#L68

**Review by tannewt — CHANGES_REQUESTED** [review]
Thanks for this! Would you mind using a subclass of OSError instead of OSError itself? That way we can differentiate a bit more. There is an example with the espidf.MemoryError already.

**Comment by jepler** [comment]
Are you thinking just one additional class for any esp error?  if not, can you let me know what classes you're looking for?

**Comment by tannewt** [comment]
@jepler Ya, one to start with in addition to the existing MemoryError. We can subclass the IDFError exception in the future if need be.

**Pull request closed without merging**